src/Microsoft.ML.Data/Data/RowCursorUtils.cs

```diff
@@ -469,6 +469,35 @@ public static IDataView RowAsDataView(IHostEnvironment env, Row row)
             return new OneRowDataView(env, row);
         }
 
+        /// <summary>
+        /// Given a collection of <see cref="Schema.Column"/>, that is a subset of the Schema of the data, create a predicate,
+        /// that when passed a column index, will return <langword>true</langword> or <langword>false</langword>, based on whether
+        /// the column with the given <see cref="Schema.Column.Index"/> is part of the <paramref name="colsNeeded"/>.
+        /// </summary>
+        /// <param name="colsNeeded">The subset of columns from the <see cref="Schema"/> that are needed from this <see cref="RowCursor"/>.</param>
+        /// <param name="sourceSchema">The <see cref="Schema"/> from where the colsNeeded originate.</param>
+        [BestFriend]
+        internal static Func<int, bool> FromColumnsToPredicate(IEnumerable<Schema.Column> colsNeeded, Schema sourceSchema)
+        {
+            Contracts.CheckValue(colsNeeded, nameof(colsNeeded));
+            Contracts.CheckValue(sourceSchema, nameof(sourceSchema));
+
+            if (colsNeeded.Count() == 0)
+                return c => false;
+
+            var indicesRequested = new HashSet<int>();
+
+            foreach (var col in colsNeeded)
+            {
+                if(sourceSchema.GetColumnOrNull(col.Name) == null)
+                    throw Contracts.Except($"The requested column: {col} is not part of the {nameof(sourceSchema)}");
+
+                indicesRequested.Add(col.Index);
+            }
+
+            return c => indicesRequested.Contains(c);
```

Comment:
the implementation on iteration 7 - 9 was different; bool[] for the indices. Switched to this because noticed that the indices of the colsNeeded in the appendedDataView of the StatefulFilterTransform, did not match the index of the same column name and type on the appended data view. Thinking back, the problem is/was in the StatefulFilterTransform.  Revert this back.